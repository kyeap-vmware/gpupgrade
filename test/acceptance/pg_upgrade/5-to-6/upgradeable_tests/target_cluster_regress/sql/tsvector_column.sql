-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that tables with tsvector columns can be upgraded.
-- pg_upgrade reference: old_8_3_rebuild_tsvector_tables
-- 8.3 sorts lexemes by its length and if lengths are the same then it uses
-- alphabetic order;  8.4 sorts lexemes in lexicographical order, e.g.
-- => SELECT 'c bb aaa'::tsvector;
--   tsvector
--   ----------------
--    'aaa' 'bb' 'c'		   -- 8.4
--    'c' 'bb' 'aaa'		   -- 8.3

-- Verify that the table is still usable after upgrade and the tsvector column is sorted correctly
-- in lexicographical order.
EXPLAIN (COSTS OFF) SELECT * FROM tsvector_table;
SELECT * FROM tsvector_table;
