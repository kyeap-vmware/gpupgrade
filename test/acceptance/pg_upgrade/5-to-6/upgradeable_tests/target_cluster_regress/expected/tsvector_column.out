-- Copyright (c) 2017-2023 VMware, Inc. or its affiliates
-- SPDX-License-Identifier: Apache-2.0

-- Test to ensure that tables with tsvector columns can be upgraded.
-- pg_upgrade reference: old_8_3_rebuild_tsvector_tables
-- 8.3 sorts lexemes by its length and if lengths are the same then it uses
-- alphabetic order;  8.4 sorts lexemes in lexicographical order, e.g.
-- => SELECT 'c bb aaa'::tsvector;
--   tsvector
--   ----------------
--    'aaa' 'bb' 'c'		   -- 8.4
--    'c' 'bb' 'aaa'		   -- 8.3

-- Verify that the table is still usable after upgrade and the tsvector column is sorted correctly
-- in lexicographical order.
EXPLAIN (COSTS OFF) SELECT * FROM tsvector_table;
 QUERY PLAN                               
------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3) 
   ->  Seq Scan on tsvector_table         
 Optimizer: Postgres query optimizer      
(3 rows)
SELECT * FROM tsvector_table;
 a  | b                          
----+----------------------------
 3  | 'aaa3':1 'bb3':2 'c3':3    
 4  | 'aaa4':1 'bb4':2 'c4':3    
 5  | 'aaa5':1 'bb5':2 'c5':3    
 6  | 'aaa6':1 'bb6':2 'c6':3    
 7  | 'aaa7':1 'bb7':2 'c7':3    
 1  | 'aaa1':1 'bb1':2 'c1':3    
 2  | 'aaa2':1 'bb2':2 'c2':3    
 8  | 'aaa8':1 'bb8':2 'c8':3    
 9  | 'aaa9':1 'bb9':2 'c9':3    
 10 | 'aaa10':1 'bb10':2 'c10':3 
(10 rows)
